src/Interactive/Features/Interactive/Core/InteractiveHost.Service.cs

```diff
@@ -208,11 +215,21 @@ private static void RunServer(string serverPort, string semaphoreName, int clien
                             ServiceName,
                             WellKnownObjectMode.Singleton);
 
-                        var uiThread = new Thread(UIThread);
-                        uiThread.SetApartmentState(ApartmentState.STA);
-                        uiThread.IsBackground = true;
-                        uiThread.Start();
-                        s_uiReady.Wait();
+                        using (var resetEvent = new ManualResetEventSlim(false))
+                        {
+                            var uiThread = new Thread(() =>
+                            {
+                                var c = new Control();
+                                c.CreateControl();
+                                s_UIThreadScheduler = TaskScheduler.FromCurrentSynchronizationContext();
```

Comment:
RunServer is called exactly once by the Main method for service, before any instances are created.